components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/MyAccountApplicationCreationListener.java

```diff
@@ -0,0 +1,193 @@
+            throw new IdentityApplicationManagementException("Error occurred while checking role existence.");
+        }
+    }
+
+    private void addApiResourceToApplication(String myAccountId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        APIResource apiResource = getImpersontionAPIResource(tenantDomain);
+
+        AuthorizedAPI authorizedAPI = new AuthorizedAPI(
+                myAccountId,
+                apiResource.getId(),
+                RBAC_AUTHORIZATION,
+                apiResource.getScopes(),
+                TENANT
+        );
+        try {
+            authorizedAPIManagementService.addAuthorizedAPI(myAccountId, authorizedAPI, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            throw new IdentityApplicationManagementException("Error occured while adding API resource.");
+        }
+    }
+
+    private boolean isAPIResourceRegistered(String myAccountId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        try {
+            List<AuthorizedAPI> authorizedAPIs = authorizedAPIManagementService.getAuthorizedAPIs(
+                    myAccountId, tenantDomain);
+            for (AuthorizedAPI authorizedAPI : authorizedAPIs) {
+                if (authorizedAPI.getAPIIdentifier().equals("system:impersonation")) {
+                    return true;
+                }
+            }
+            return false;
+        } catch (IdentityApplicationManagementException e) {
+            throw new IdentityApplicationManagementException("Error occured while adding API resource.");
+        }
+    }
+
+    private APIResource getImpersontionAPIResource(String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        APIResourceManager apiResourceManager = ApplicationManagementServiceComponentHolder.getInstance()
+                .getAPIResourceManager();
+        APIResource apiResource = null;
+        try {
+            apiResource = apiResourceManager.getAPIResourceByIdentifier("system:impersonation",
+                    tenantDomain);
+        } catch (APIResourceMgtException e) {
+            throw new IdentityApplicationManagementException("Error occurred while retrieving API resource.");
+        }
+        if (apiResource == null) {
+            throw new IdentityApplicationManagementException("Api resources unavailable.");
+        }
+        return apiResource;
+    }
+
+    private String getMyAccountAppId(String tenantDomain) throws IdentityApplicationManagementException {
+
+        ApplicationManagementService applicationManagementService = ApplicationManagementService.getInstance();
+        try {
+            String appUUID = applicationManagementService.getApplicationUUIDByName(
+                    MY_ACCOUNT_APPLICATION_NAME, tenantDomain);
+            if (StringUtils.isBlank(appUUID)) {
+                throw new MigrationClientException("ApplicationUUID is null.");
```

Comment:
Is this exception correct?

Reply:
Addressed with 82654fc5ad702dedb715d648bde39c8f8b15aa16